tests/qiskit_convert_test.py

```diff
@@ -961,3 +961,42 @@ def test_unitary_gate() -> None:
     assert cmds[0].op.type == OpType.Unitary1qBox
     assert cmds[1].op.type == OpType.Unitary2qBox
     assert cmds[2].op.type == OpType.Unitary3qBox
+
+
+def test_ccz_conversion() -> None:
+    qc_ccz = QuantumCircuit(4)
+    qc_ccz.append(qiskit_gates.CCZGate(), [0, 1, 2])
+    qc_ccz.append(qiskit_gates.CCZGate(), [3, 1, 0])
+    tkc_ccz = qiskit_to_tk(qc_ccz)
+    assert tkc_ccz.n_gates_of_type(OpType.CnZ) == tkc_ccz.n_gates == 2
+    # bidirectional CnZ conversion already supported
+
+
+def test_csx_conversion() -> None:
+    qc_csx = QuantumCircuit(2)
+    qc_csx.append(qiskit_gates.CSXGate(), [0, 1])
+    qc_csx.append(qiskit_gates.CSXGate(), [1, 0])
+    converted_tkc = qiskit_to_tk(qc_csx)
+    assert converted_tkc.n_gates == 2
+    assert converted_tkc.n_gates_of_type(OpType.CSX) == 2
+    u1 = converted_tkc.get_unitary()
+    new_tkc_csx = Circuit(2)
+    new_tkc_csx.add_gate(OpType.CSX, [0, 1]).add_gate(OpType.CSX, [1, 0])
+    u2 = new_tkc_csx.get_unitary()
+    assert compare_unitaries(u1, u2)
+    converted_qc = tk_to_qiskit(new_tkc_csx)
+    assert converted_qc.count_ops()["csx"] == 2
+    qc_c3sx = QuantumCircuit(4)
+    qc_c3sx.append(qiskit_gates.C3SXGate(), [0, 1, 2, 3])
+    tkc_c3sx = qiskit_to_tk(qc_c3sx)
```

Comment:
Is it possible to convert this back to qiskit? Could we start with a tket circuit and convert it to qiskit and back and check if they are still identical?

Reply:
It is possible in theory but we would have to add extra logic for that.

We could add a check to `tk_to_qiskit` that if it finds a `QControlBox` that checks the base gate of the QControl and then converts it to a qiskit gate with the right number of controls.

Problem is I think we'd have to somehow store the unsupported qiskit `ControlGate`s somewhere to do this.

Reply:
Ha okay, in that case it might be better not to do that